### PR TITLE
cmake: fix support for je/mi malloc and the mallocallocator

### PR DESCRIPTION
**Pull request by cmcfarlen**
fixes [#10377](https://github.com/apache/trafficserver/issues/10377)

### CODE REVIEW

**Review by JosiahWI — APPROVED** [review]
Looks good!

**Review by JosiahWI — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -229,7 +230,7 @@ if(TS_HAS_JEMALLOC AND TS_HAS_MIMALLOC)
 endif()
 
 if(TS_HAS_JEMALLOC)
-    link_libraries(jemalloc)
+    link_libraries(jemalloc::jemalloc)
 elseif(TS_HAS_MIMALLOC)
     link_libraries(mimalloc)
```
Comment (JosiahWI) [inline CMakeLists.txt:235]:
```suggestion
    link_libraries(mimalloc::mimalloc)
```
here too

**Review by JosiahWI — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (cmcfarlen) [reply CMakeLists.txt:235]: There isn't currently a `mimalloc::mimalloc` target

**Comment by bryancall** [comment]
Testing malloc:
```
$ cmake -B build -G Ninja -DCMAKE_INSTALL_PREFIX=/opt/ats -DCMAKE_BUILD_TYPE=Debug -DENABLE_AUTEST=ON -DBUILD_EXPERIMENTAL_PLUGINS=ON -DENABLE_MALLOC_ALLOCATOR=ON
$ cd build; ninja
$ grep -ir TS_USE_MALLOC_ALLOCATOR
include/tscore/ink_config.h:#define TS_USE_MALLOC_ALLOCATOR 1
```

**Comment by bryancall** [comment]
By default it looks like we enable jemalloc.  I don't believe this is what we do for autotools:
```
CMakeCache.txt:ENABLE_JEMALLOC:STRING=AUTO
CMakeCache.txt:jemalloc_INCLUDE_DIR:PATH=/usr/include/jemalloc
CMakeCache.txt:jemalloc_LIBRARY:FILEPATH=/usr/lib64/libjemalloc.so
```

File: include/tscore/ink_memory.h [hunk 2]
```diff
@@ -49,7 +49,11 @@
 #endif
 
 #if TS_HAS_JEMALLOC
+#if __has_include(<jemalloc/jemalloc.h>)
 #include <jemalloc/jemalloc.h>
+#else
+#include <jemalloc.h>
```
Comment (bryancall) [inline include/tscore/ink_memory.h:55]: Ee only need to have `#include <jemalloc.h>` in the code for compiling with `cmake`.    The include directory detection add in the subdir `jemalloc`.

However, with autotools we need to have the `#include <jemalloc/jemalloc.h>`

**Review by bryancall — APPROVED** [review]

**Pull request merged**